code/datums/flock/flock.dm

```diff
@@ -347,13 +347,14 @@
 /datum/flock/proc/registerUnit(var/atom/movable/D)
 	if(isflock(D))
 		src.units |= D
+	D.AddComponent(/datum/component/flock_interest, src)
 	var/datum/abilityHolder/flockmind/aH = src.flockmind.abilityHolder
 	aH.updateCompute()
 
 /datum/flock/proc/removeDrone(var/atom/movable/D)
 	if(isflock(D))
 		src.units -= D
-
+		D.GetComponent(/datum/component/flock_interest)?.RemoveComponent(/datum/component/flock_interest)
```

Comment:
There are a couple of other instances where `RemoveComponent` is called with the type of the component, but it doesn't seem to ever actually do anything with the type, so I'm really not sure why?

Reply:
No idea why it's required. I'm pretty sure it doesn't compile if you don't do it.